cmd/deployment.go

```diff
@@ -134,9 +136,9 @@ func newDeploymentCreateCmd(out io.Writer) *cobra.Command {
 	}
 
 	var nfsMountDAGDeploymentEnabled, triggererEnabled, gitSyncDAGDeploymentEnabled bool
-	appConfig, err := houstonClient.GetAppConfig()
+	appConfig, err := deployment.GetAppConfig(houstonClient)
 	if err != nil {
-		fmt.Println("Error checking feature flag", err)
+		initDebugLogs = append(initDebugLogs, fmt.Sprintf("Error checking feature flag: %s", err.Error()))
```

Comment:
Just realised, storing logs in memory can lead to other folks misusing it, and also increasing memory footprint of the CLI, without us having direct control over it, as some of these are errors are being thrown from external packages.

I think its not a bad idea to simply convert this to `logrus.Debugln` like we have done it for other similar error in this PR. These error checks where simply added as linter was recommending for it, and wasn't a request from customer end, and in case these logs are needed then one would be still be able to access it, by setting the `verbosity` as `debug` in the config.

Reply:
The problem is, as discussed on Slack, this line of code here is executed before the `setupLogs` statement in PersistentPrerunE calls.

- CLI Initializes
- Get App Config
- Check Feature flags
- Add flags on the CLI if feature flag enabled in App Config
- Execute root command's PersistentPreRunE
- Execute the command

If there is an error in the `Get App Config`, the `logrus.Debugln` will not be printed when user specifies `--verbosity=debug` since root's PersistentPreRunE (and SetUpLogs) has not been executed yet. This is why we need to "buffer" debug logs produced before CLI initialization.

In order to fix the "memory footprint", I can just delete the slice from memory after the printDebugLogs occured.

Reply:
cool 👍, also it would be great if we can add a comment around the `initDebugLogs` var (https://github.com/astronomer/astro-cli/pull/502/files#diff-ab967ab1a2f3a1b769106eeb7bfe892ef0e81d1d27811fa15be08e6749feee1fR22), so that in future folks don't end up misinterpreting the usecase.